packages/devextreme/testing/tests/DevExpress.ui.widgets/chat.tests.js

```diff
@@ -347,6 +352,138 @@ QUnit.module('renderMessage', moduleConfig, () => {
+
+        const $textArea = $element.find(`.${TEXTEDITOR_INPUT_CLASS}`);
+        const $button = $element.find(`.${CHAT_MESSAGE_BOX_BUTTON_CLASS}`);
+
+        keyboardMock($textArea).focus().type('new text message');
+
+        $button.trigger('dxclick');
+
+        assert.strictEqual(onMessageSend.callCount, 1);
+    });
+
+    QUnit.test('New message should be created after clicking the send button if there is text', function(assert) {
+        const text = 'new text message';
+
+        const $textArea = this.$element.find(`.${TEXTEDITOR_INPUT_CLASS}`);
+        const $button = this.$element.find(`.${CHAT_MESSAGE_BOX_BUTTON_CLASS}`);
+
+        keyboardMock($textArea).focus().type(text);
+
+        $button.trigger('dxclick');
+
+        const $bubbles = this.$element.find(`.${CHAT_MESSAGE_BUBBLE_CLASS}`);
+        const bubble = $bubbles[$bubbles.length - 1];
+
+        assert.strictEqual($(bubble).text(), text);
+    });
+
+    QUnit.test('TextArea text should be empty after clicking the send button if there is text', function(assert) {
+        const text = 'new text message';
+
+        const $textArea = this.$element.find(`.${TEXTEDITOR_INPUT_CLASS}`);
+        const $button = this.$element.find(`.${CHAT_MESSAGE_BOX_BUTTON_CLASS}`);
+
+        keyboardMock($textArea).focus().type(text);
+
+        $button.trigger('dxclick');
+
+        assert.strictEqual(this.$element.find(`.${TEXTEDITOR_INPUT_CLASS}`).get(0).value, '');
+    });
+
+    QUnit.test('onMessageSend should be called after clicking the send button if there is text', function(assert) {
+        const onMessageSend = sinon.spy();
+
+        this.instance.option({ onMessageSend });
+
+        const text = 'new text message';
+
+        const $textArea = this.$element.find(`.${TEXTEDITOR_INPUT_CLASS}`);
+        const $button = this.$element.find(`.${CHAT_MESSAGE_BOX_BUTTON_CLASS}`);
+
+        keyboardMock($textArea).focus().type(text);
+
+        $button.trigger('dxclick');
+
+        assert.strictEqual(onMessageSend.callCount, 1);
+    });
+
+    QUnit.test('onMessageSend should be get correct object after clicking the send button if there is text', function(assert) {
+        assert.expect(5);
+
+        const text = 'new text message';
+
+        const $textArea = this.$element.find(`.${TEXTEDITOR_INPUT_CLASS}`);
+        const $button = this.$element.find(`.${CHAT_MESSAGE_BOX_BUTTON_CLASS}`);
+
+        this.instance.option({
+            onMessageSend: ({ component, element, event, message }) => {
+                assert.strictEqual(component, this.instance, 'component field is correct');
+                assert.strictEqual(isRenderer(element), !!config().useJQuery, 'element is correct');
+                assert.strictEqual($(element).is(this.$element), true, 'element field is correct');
+                assert.strictEqual(event.target, $button.get(0), 'event field is correct');
+                assert.strictEqual(message.text, text, 'message field is correct');
```

Comment:
Seems, this does not match the description of our current types. Text property is inside event data.
Also we don't check timestamp, author and typing properties.

Reply:
> Also we don't check timestamp, author and typing properties

We checked it in another test

Reply:
> Seems, this does not match the description of our current types. Text property is inside event data.

We're going to update .d.ts later. We have a separate task for that